### DISCUSSION TITLE
Can't pass lexer option to colorize plugin (using chroma)

### DISCUSSION THREAD

**Discussion by apolopena** [ohmyzsh/ohmyzsh#13527] [Help] [answered]
**Describe the bug**
I am using **chroma** with the Colorize plugin and I want to use the aliases that the Colorize plugin provides: `ccat` (for `colorize_cat`) and `cless` (for `colorize_less`) to display syntax highlighting for files with no extension, based on the content of the file. 

**To Reproduce**
contents of file 'zman' is:
```
errcode=0
if [ $# -eq 0 ]
then
    >&2 echo "Not enough arguments";
    errcode=2
fi
if [ $# -eq 1 ]
then
    PAGER="less -g -s '+/^       "$1"'" command man zshall
fi
if [ $# -eq 2 ]
then
    PAGER="less -g -s '+/^       "$1"'" command man "$2"
fi
if [ $# -gt 2 ]
then
    >&2 echo "Too many arguments"
    errcode=2
fi
if [ $errcode -gt 0 ]
then
    $(exit $errcode)
fi
# vim:ft=zsh:et
```
run the command:
`ccat zman`

**Result**
Output is not colorized

**Expected behavior**
Output should be colorized using the "Bash" lexer

**Desktop**
 - OS / Distro: Microsoft Windows 10 Pro (64-bit), Version 10.0.18362
 - Latest ohmyzsh Update?: Yes
 - ZSH Version: zsh 5.8 (x86_64-ubuntu-linux-gnu)
 - Terminal emulator: Windows Terminal

**Additional context**
I assume what I am asking should be possible since I found this in the `colorize_cat` function `colorize.plugin.zsh` :
```
    # Guess lexer from file extension, or guess it from file contents if unsuccessful.
    local FNAME lexer
    for FNAME in "$@"; do
        if [[ "$ZSH_COLORIZE_TOOL" == "pygmentize" ]]; then
            lexer=$(pygmentize -N "$FNAME")
            if [[ $lexer != text ]]; then
                pygmentize -O style="$ZSH_COLORIZE_STYLE" -l "$lexer" "$FNAME"
            else
                pygmentize -O style="$ZSH_COLORIZE_STYLE" -g "$FNAME"
            fi
        else
            chroma --style="$ZSH_COLORIZE_STYLE" "$FNAME"
        fi
    done
```
I can get the example file `zman` to have colorized output by using chroma manually like this:
`chroma zman --lexer="Bash" --style="swapoff" --formatter="terminal256"`

**Comment by mcornella** [comment, marked as answer]
The `colorize_cat` plugin iterates over the arguments passed as if they are files (because `pygmentize` only supports one file at a time). This means that even if you pass `--lexer=bash`, it'll interpret that as a filename. `chroma` does support multiple files at once, so we could fix the plugin for this case.

In the meantime, you can instead pass the file as redirected input and call `ccat --lexer=bash < filename`.

**Comment by apolopena** [comment]
Ah thank you @mcornella
